internal/cmd/dev.go

```diff
@@ -0,0 +1,205 @@
+	Results resultSet `json:"results"`
+}
+
+type errorResponse struct {
+	Error struct {
+		Message string `json:"message"`
+	} `json:"error"`
+}
+
+type statement struct {
+	Query  string `json:"q"`
+	Params []interface{}
+}
+
+func (s statement) String() string {
+	var buf bytes.Buffer
+	buf.WriteString(s.Query)
+	buf.WriteString("; ")
+	jsonData, err := json.Marshal(s.Params)
+	if err != nil {
+		panic("can't convert json data from params")
+	}
+	buf.WriteString(string(jsonData))
+	return buf.String()
+}
+
+type statementList struct {
+	Statements []statement `json:"statements"`
+}
+
+func (s *statement) UnmarshalJSON(data []byte) error {
+	var query interface{}
+	err := json.Unmarshal(data, &query)
+	if err != nil {
+		return err
+	}
+
+	switch q := query.(type) {
+	case string:
+		s.Query = q
+	case map[string]interface{}:
+		qValue, ok := q["q"].(string)
+		if !ok {
+			return fmt.Errorf("invalid JSON: 'q' field is missing or not a string")
+		}
+		s.Query = qValue
+
+		switch p := q["params"].(type) {
+		case []interface{}:
+			s.Params = p
+		case map[string]interface{}:
+			var named []interface{}
+			for k, v := range p {
+				named = append(named, sql.Named(k, v))
+			}
+			s.Params = named
+		}
+	}
+
+	return nil
+}
+
+func errorJson(e error) []errorResponse {
+	return []errorResponse{
+		{
+			Error: struct {
+				Message string `json:"message"`
+			}{
+				Message: e.Error(),
+			},
+		},
+	}
+}
+
+var devCmd = &cobra.Command{
+	Use:               "dev",
+	Short:             "starts a development server for Turso",
+	Args:              cobra.NoArgs,
+	ValidArgsFunction: noFilesArg,
+	RunE: func(cmd *cobra.Command, args []string) error {
+		cmd.SilenceUsage = true
+		port := fmt.Sprintf(":%d", devPort)
+		var filename string
+		var dbLocation string
+		if devFile == "" {
+			filename = "file:somefile.db"
+			devFile = ":memory:"
+			dbLocation = "in memory"
+		} else {
+			filename = fmt.Sprintf("file:%s", devFile)
+			dbLocation = fmt.Sprintf("at %s", devFile)
+		}
+
+		fmt.Printf("For environments where a filesystem is available:\n")
+		fmt.Printf("%s use %s as a URL. This server is not needed. (Ctrl-C)\n", internal.Emph("→  "), internal.Emph(filename))
+		fmt.Printf("For all others environments:\n")
+		fmt.Printf("%s connect to %s%s. Database is %s\n", internal.Emph("→  "), internal.Emph("http://localhost"), internal.Emph(port), internal.Emph(dbLocation))
+		fmt.Printf("%s In particular, the following Turso features are currently not present in local-server mode:\n", internal.Warn("This is experimental:"))
+		fmt.Printf("  * extensions.\n")
+		fmt.Printf("  * interactive transactions/libsql-based URLs.\n")
+
+		gin.SetMode(gin.ReleaseMode)
+		r := gin.New()
+		if !verboseFlag {
+			r.Use(gin.LoggerWithWriter(io.Discard))
+		}
+
+		db, err := sql.Open("sqlite3", devFile)
+		if err != nil {
+			fmt.Println("Failed to connect to SQLite database:", err)
+			return err
+		}
+		defer db.Close()
+
+		r.POST("/", func(c *gin.Context) {
```

Comment:
shouldn't this be /queries?

Reply:
https://github.com/honzasp/sqld/blob/hrana2/docs/http_api.md

Reply:
I don't think so, at least this works now with http as is.